File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack = []
    for i in range(len(s)) :
        if s[i] == '(' :
            stack.append(s[i])
        elif s[i] == ')':
            if len(stack) == 0:
                return False
            if stack[-1] == ')':
                stack.append(s[i])
            elif stack[-1] == '(':
                stack.pop()
    if len(stack) > 0:
        return False
    
    return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let answer = true;
    let stack = new Array();

    for (let i = 0; i < s.length; i++) {
        if (s[i] === "(") {
            stack.push("(")
        } else if (s[i] === ")") {
            if (stack.length < 1) {
                return false
            }
            else if (stack.length > 0) {
                if (stack[stack.length - 1] === '(') {
                    stack.pop()
                }
                else {
                    return false
                }
            }
        }
    }
    
    return stack.length > 0 ? false : true;
}